LibSerialize.lua

```diff
@@ -1082,7 +1105,7 @@ LibSerialize._WriterTable = {
             local mapCount = 0
             local entireMapSerializable = true
             for k, v in pairs(tab) do
-                local isArrayKey = type(k) == "number" and k >= 1 and k <= arrayCount and not IsFractional(k)
+                local isArrayKey = type(k) == "number" and k >= 1 and k <= arrayCount and not IsFloatingPoint(k)
```

Comment:
Mind refactoring this into a small helper function, since it's repeated three times? You didn't introduce it, but would be nice to clean up.